[TSP Migration]--datafactory

welovej opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [ ] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [ ] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

welovej committed:
converted

welovej committed:
could compile

welovej committed:
add suppress

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Update tspconfig.yaml

welovej committed:
remove operation id

welovej committed:
justification

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
update

welovej committed:
Update models.tsp

welovej committed:
Revert "Update models.tsp"

This reverts commit 2694642edfbbbd3d28bb1eb0bdc4ea969c3b20a3.

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-DataFactory

welovej committed:
Merge remote-tracking branch 'upstream/main' into TspMig-DataFactory

welovej committed:
Update models.tsp

welovej committed:
Update models.tsp

welovej committed:
Update models.tsp

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
update

v-jiaodi committed:
Merge branch 'main' into TspMig-DataFactory

jliusan committed:
update config

ChenxiJiang333 committed:
Create client.tsp

welovej committed:
update

welovej committed:
update

welovej committed:
Update Factory.tsp

welovej committed:
update

welovej committed:
update

welovej committed:
Update back-compatible.tsp

jliusan committed:
add go config

welovej committed:
update

ChenxiJiang333 committed:
Update client.tsp

v-jiaodi committed:
update client name

v-jiaodi committed:
Merge branch 'TspMig-DataFactory' of https://github.com/welovej/azure-rest-api-specs into pr/38941

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Update models.tsp

v-huizhu2 committed:
Enable use-object-for-unknown in tspconfig.yaml

v-huizhu2 committed:
Add Java client names for various services

ChenxiJiang333 committed:
Update client.tsp

welovej committed:
remove redundant model Resource

v-huizhu2 committed:
Add Java compatibility annotations for resources

msyyc committed:
Merge branch 'main' into TspMig-DataFactory

v-huizhu2 committed:
Add allowHostnameCNMismatch and other properties

v-huizhu2 committed:
Update client names to use Azure.ResourceManager

welovej committed:
properties to be required

welovej committed:
fix go breaking

welovej committed:
update

welovej committed:
update

welovej committed:
update

welovej committed:
Update client.tsp

v-huizhu2 committed:
Update tspconfig.yaml with new settings

Set uuid-as-string and float32-as-double to false.

jliusan commented:
/azp run SDK Validation - Go

v-huizhu2 committed:
Add client names for various SSIS activities

v-huizhu2 committed:
Add DataLakeAnalyticsUSQLActivityTypeProperties and SSISLogLocationTypeProperties

v-jiaodi commented on specification/datafactory/resource-manager/Microsoft.DataFactory/DataFactory/stable/2018-06-01/datafactory.json:
```diff
@@ -1,8328 +0,0 @@
-            "name": "Type",
-            "modelAsString": true
-          }
-        },
```
why change type to :
```
"type": {
          "$ref": "#/definitions/LinkedServiceReferenceType",
          "description": "Linked service reference type."
        }
"LinkedServiceReferenceType": {
      "type": "string",
      "description": "Linked service reference type.",
      "enum": [
        "LinkedServiceReference"
      ],
      "x-ms-enum": {
        "name": "LinkedServiceReferenceType",
        "modelAsString": true,
        "values": [
          {
            "name": "LinkedServiceReference",
            "value": "LinkedServiceReference",
            "description": "LinkedServiceReference"
          }
        ]
      }
    }
```

welovej replied:
Fixed

welovej committed:
update

v-jiaodi committed:
Merge branch 'main' into TspMig-DataFactory

v-huizhu2 committed:
Remove uuid-as-string option from tspconfig.yaml

welovej committed:
fix

welovej committed:
fix

welovej commented on specification/datafactory/DataFactory.Management/models.tsp:
Fixed

v-huizhu2 commented on specification/datafactory/DataFactory.Management/models.tsp:
Can change @format("uuid") to uuid type here ?

welovej committed:
fix

welovej committed:
fix

welovej committed:
fix

v-huizhu2 committed:
Add usage and access annotations for translators

v-huizhu2 committed:
Remove Java annotations for CopyTranslator and others

Removed Java usage and access annotations for several translators and settings.

welovej committed:
update

v-huizhu2 committed:
Add usage and access annotations for translators

v-huizhu2 committed:
Add client names for FactoryVSTSConfiguration and SalesforceV2Source

v-xuto committed:
Refactor folder

jliusan committed:
add go clientName

welovej committed:
fix

jliusan committed:
add go clientName

pshao25 committed:
Merge branch 'main' into TspMig-DataFactory
